misk/src/main/kotlin/misk/environment/Environment.kt

```diff
@@ -4,39 +4,55 @@ import com.google.common.base.Preconditions
 import misk.logging.getLogger
 
 /** The environment in which the application is running */
+@Deprecated("use Deployment instead")
 enum class Environment {
   TESTING,
   DEVELOPMENT,
   STAGING,
   PRODUCTION;
 
+  fun isFake(): Boolean = this == TESTING || this == DEVELOPMENT
+
+  fun isReal(): Boolean = !isFake()
+
   companion object {
     internal val logger = getLogger<Environment>()
     private const val ENV_ENVIRONMENT = "ENVIRONMENT"
-    private lateinit var env: Environment
+
+    private lateinit var rawEnv: String
 
     fun setTesting() {
-      if (::env.isInitialized) {
-        Preconditions.checkState(env == TESTING, "Environment already set to ${env.name}")
+      if (::rawEnv.isInitialized) {
+        Preconditions.checkState(rawEnv == TESTING.name, "Environment already set to $rawEnv")
       }
-      env = TESTING
+      rawEnv = TESTING.name
     }
 
     @JvmStatic
     fun fromEnvironmentVariable(): Environment {
+      return valueOf(when (val rawEnv = rawEnvironment()) {
```

Comment:
this line has a lot of density to it. Maybe separate string mapping from valueOf ?